examples/core/core_2d_camera_smooth_pixelperfect.c

```diff
@@ -0,0 +1,142 @@
+    //--------------------------------------------------------------------------------------
+
+    // Main game loop
+    while (!WindowShouldClose())    // Detect window close button or ESC key
+    {
+        // Update
+        //----------------------------------------------------------------------------------
+        rotation++; // Rotate the rectangles.
+
+        // Make the camera move to demonstrate the effect.
+        cameraX = (sin(GetTime()) * 50.0f) - 10.0f;
+        cameraY = cos(GetTime()) * 30.0f;
+
+        screenSpaceCamera.target = (Vector2){ cameraX, cameraY };
+
+        if (screenSpaceCamera.target.x >= 1 || screenSpaceCamera.target.x <= -1) // Round worldCamera's X, keep the decimals on screenSpaceCamera.
+        {
+            worldSpaceCamera.target.x = (int)screenSpaceCamera.target.x;
+            screenSpaceCamera.target.x -= worldSpaceCamera.target.x;
+            screenSpaceCamera.target.x *= virtualRatio;
+        }
+
+        if (screenSpaceCamera.target.y >= 1 || screenSpaceCamera.target.y <= -1) // Round worldCamera's Y, keep the decimals on screenSpaceCamera.
+        {
+            worldSpaceCamera.target.y = (int)screenSpaceCamera.target.y;
+            screenSpaceCamera.target.y -= worldSpaceCamera.target.y;
+            screenSpaceCamera.target.y *= virtualRatio;
+        }
+
+        //----------------------------------------------------------------------------------
+
+        // Draw
+        //----------------------------------------------------------------------------------
+        BeginDrawing();
+        ClearBackground(RED); // This is for debug purposes. If you see red, then you've probably done something wrong.
+
+        BeginTextureMode(renderTexture);
+
+        BeginMode2D(worldSpaceCamera);
+
+        ClearBackground(RAYWHITE); // This is the color you should see as background color.
+
+        // Draw the rectangles
+        DrawRectanglePro(firstRectangle, origin, rotation, BLACK);
+        DrawRectanglePro(secondRectangle, origin, -rotation, RED);
+        DrawRectanglePro(thirdRectangle, origin, rotation + 45.0f, BLUE);
+
+        EndMode2D();
+
+        EndTextureMode();
+
+        BeginMode2D(screenSpaceCamera);
+
+        // Draw the render texture with an offset of 1 worldSpace unit/pixel, so that the content behind the renderTexture is not shown.
+        // Also the renderTexture's height is flipped (in the source Rectangle), due to OpenGL reasons.
+        DrawTexturePro(
+            renderTexture.texture,
+            (Rectangle)
+        {
+            0.0f, 0.0f, (float)renderTexture.texture.width, (float)-renderTexture.texture.height
+        },
+            (Rectangle)
+        {
+            -virtualRatio, -virtualRatio, screenWidth + (virtualRatio * 2), screenHeight + (virtualRatio * 2)
+        },
+                (Vector2)
+            {
+                0.0f, 0.0f
+            },
+                0.0f,
+                    WHITE
+                    );

```

Comment:
I would precompute these rectangles and clean up the whitespace you've got going on.